rgw/v2/tests/s3_swift/test_multisite_syncpolicy_prefix_tag.py

```diff
@@ -0,0 +1,182 @@
+"""
+Usage: test_syncpolicy_prefix_tag.py -c <input_yaml>
+
+<input_yaml>
+    Note: Following yaml can be used
+    configs/test_syncpolicy_prefix_tag.yaml
+
+Operation:
+a. On a MS setup, create sync policy with configuration to sync objects having a prefix or tag or both.
+b. Write objects having said prefix or tag as mentioned in the config.
+"""
+
+
+import argparse
+import json
+import logging
+import os
+import random
+import sys
+import time
+import traceback
+
+sys.path.append(os.path.abspath(os.path.join(__file__, "../../../..")))
+import v2.lib.resource_op as s3lib
+from v2.lib import resource_op
+from v2.lib.exceptions import RGWBaseException, TestExecError
+from v2.lib.s3.auth import Auth
+from v2.lib.s3.write_io_info import BasicIOInfoStructure, IOInfoInitialize
+from v2.tests.s3_swift import reusable
+from v2.utils import utils
+from v2.utils.log import configure_logging
+from v2.utils.test_desc import AddTestInfo
+
+log = logging.getLogger(__name__)
+TEST_DATA_PATH = None
+
+
+def test_exec(config, ssh_con):
+    """
+    Executes test based on configuration passed
+    Args:
+        config(object): Test configuration
+    """
+    io_info_initialize = IOInfoInitialize()
+    basic_io_structure = BasicIOInfoStructure()
+    io_info_initialize.initialize(basic_io_structure.initial())
+    user_info = s3lib.create_users(config.user_count)
+
+    # Create Zonegroup policy
+    log.info("Creating Zone group policy at Enabled state")
+    group_id = "global_group"
+    reusable.group_operation(group_id, "create", "enabled")
+    reusable.flow_operation(group_id, "create")
+    reusable.pipe_operation(group_id, "create")
+
+    for user in user_info:
+        auth = Auth(user, ssh_con, ssl=config.ssl)
+        rgw_conn = auth.do_auth()
+
+        for bc in range(config.bucket_count):
+            bucket_name = utils.gen_bucket_name_from_userid(user["user_id"], rand_no=bc)
+            bucket = reusable.create_bucket(bucket_name, rgw_conn, user)
+            log.info(f"Bucket {bucket_name} created")
+            prefix = "foo"
+            obj_tag = '{"TagSet":[{"Key":"colour", "Value":"red"}]}'
+            # Create bucket sync policy
+            group_id1 = "group-" + bucket_name
+            reusable.group_operation(
+                group_id1,
+                "create",
+                "enabled",
+                bucket_name,
+            )
+            if config.test_ops["has_prefix"] and config.test_ops["has_tag"]:
+                detail = f" --prefix={prefix} --tags-add={obj_tag}"
+            elif (
+                config.test_ops["has_prefix"] is True
+                and config.test_ops["has_tag"] is False
+            ):
+                detail = f" --prefix={prefix}"
+            elif (
+                config.test_ops["has_prefix"] is False
+                and config.test_ops["has_tag"] is True
+            ):
+                detail = f" --tags-add={obj_tag}"
```

Comment:
could you simplify these lines as discussed

Reply:
thanks @hmaheswa , updated